rake_config/vanishing_data.rb

```diff
@@ -0,0 +1,74 @@
+class SourceCSV
+  def initialize(str)
+    @raw = str
+  end
+
+  def ids
+    grouped.keys
+  end
+
+  def rows(*kk)
+    kk.map { |k| grouped[k].first }
+  end
+
+  private
+
+  attr_reader :raw
+
+  def as_csv
+    @csv ||= CSV.parse(raw, headers: true, header_converters: :symbol)
+  end
+
+  def grouped
+    @grouped ||= as_csv.group_by { |r| r[:id] }
+  end
+end
+
+class SourceHistory
+  # Given a Source file, look at the prior version in git history and
+  # provide access to information that has vanished since then
+  def initialize(source)
+    @source = source
+  end
+
+  def vanishing_data
+    old.rows(*vanishing_ids)
+  end
+
+  private
+
+  attr_reader :source
+
+  def repo_root
+    Pathname.new('../../..').realpath
+  end
+
+  def full_pathname
+    source.pathname.realpath.relative_path_from(repo_root)
+  end
+
+  def old
+    @old ||= SourceCSV.new(`git show @~1:#{full_pathname}`)
```

Comment:
I can't immediately think of a better way to execute this command without shelling out to git. However I do think it would be clearer if the command used `HEAD` instead of `@` since I wasn't aware `@` was an alias for `HEAD` until I read `git help revisions` just now!

```ruby
`git show HEAD~1:#{full_pathname}`
```